python/TestHarness/unit_tests/test_HarnessTester.py

```diff
@@ -95,3 +95,41 @@ def testDuplicateOutputsOK(self):
         self.assertNotRegexpMatches(output, 'heavy_out.e')
         # all
         self.assertNotRegexpMatches(output, 'FATAL TEST HARNESS ERROR')
+
+    def testDeleted(self):
+        """
+        Test that deleted tests returns a failed deleted test when extra info argument is supplied
+        """
+        with self.assertRaises(subprocess.CalledProcessError) as cm:
+            self.runTests('-i', 'deleted', '-e')
+
+        e = cm.exception
+        self.assertRegexpMatches(e.output, 'test_harness\.deleted.*?deleted \(test deleted test\)')
+
+
+    def testDryRun(self):
+        """
+        Test that --dry-run returns a passing status
+        """
+        output = self.runTests('-i', 'diffs', '--dry-run')
+
+        self.assertRegexpMatches(output, 'test_harness\.exodiff.*?DRY RUN')
+        self.assertRegexpMatches(output, 'test_harness\.csvdiff.*?DRY RUN')
+
+        # Skipped caveat test which returns skipped instead of 'DRY RUN'
+        output = self.runTests('-i', 'depend_skip_tests', '--dry-run')
+        self.assertIn('skipped (always skipped)', output)
+
+        # NOTE: This test normally returns (skipped dependency). However
+        # with dry run, the TestHarness has no idea that this is the case
+        # because in order to figure that out, the test has to 'run' and we are
+        # not running any tests (its a dry run after all)!
+        self.assertRegexpMatches(output, 'test_harness\.needs_a.*?DRY RUN')
+
+        # Deleted caveat test which returns a deleted failing tests while
+        # performing a dry run
+        with self.assertRaises(subprocess.CalledProcessError) as cm:
+            self.runTests('-i', 'deleted', '-e', '--dry-run')
+
+        e = cm.exception
+        self.assertRegexpMatches(e.output, 'test_harness\.deleted.*?deleted \(test deleted test\)')
```

Comment:
OK - we can fix this later. Deleted tests really should be treated the same as skipped. This isn't your problem, it's just a design issue as you've pointed out.